llvm/lib/RemoteCachingService/Client/Client.cpp

```diff
@@ -482,31 +482,34 @@ class CASDBClientImpl : public CASDBClient {
 
 } // namespace
 
-Expected<ClientServices>
-cas::remote::createCompilationCachingRemoteClient(StringRef SocketPath) {
+static std::shared_ptr<grpc::Channel> createGRPCChannel(StringRef SocketPath) {
   std::string Address("unix:");
   Address += SocketPath;
-  const auto Channel = grpc::CreateChannel(std::move(Address),
-                                           grpc::InsecureChannelCredentials());
+  grpc::ChannelArguments Args{};
+  // Remove the initial connection timeout. If the execution environment gets
+  // into a state where the processes don't make progress, for a period longer
+  // than the timeout, it can cause unnecessary compilation errors if the
+  // connection fails due to the timeout. Note that the connection still fails
+  // if the socket path does not exist or there is no process listening on it.
+  Args.SetInt(GRPC_ARG_MIN_RECONNECT_BACKOFF_MS, INT_MAX);
```

Comment:
Is this a client side only parameter or does it matter for server side?

Reply:
Is your question whether a gRPC server would also need to set this option? AFAICT just changing the clang client seems to address the originating issue.

Reply:
Yeah, I mean does the server need to set this option too. My worry is that if the client (clang) is hanging and suspended, the server is going to error out and not recoverable. If that is not a worry, it is also fine.